src/NServiceBus.Persistence.DynamoDB/Outbox/PartialOutboxResultException.cs

```diff
@@ -0,0 +1,30 @@
+namespace NServiceBus.Persistence.DynamoDB;
+
+using System;
+using System.Runtime.Serialization;
+
+/// <summary>
+/// Exception that can occur when the outbox records are queried immediately after the outbox transaction committed due to the read-committed transaction isolation levels between transactional write items and query operations.
+///
+/// For more information consult https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-apis.html#transaction-isolation.
+///
+/// Retrying should resolve this problem.
+/// </summary>
+[Serializable]
+public class PartialOutboxResultException : Exception
+{
+    /// <summary>
+    /// Initializes a new instance of <see cref="PartialOutboxResultException" />.
+    /// </summary>
+    public PartialOutboxResultException(string messageId, int transportOperationsRetrieved, int expectedNumberOfTransportOperations) : base(
+        $"Partial outbox results retrieved with {transportOperationsRetrieved} instead of {expectedNumberOfTransportOperations} transport operation while attempting to load the outbox records for the message id '{messageId}'. This problem can occur due to read-committed isolation levels between transactional write items and query operations. For more information consult https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-apis.html#transaction-isolation")
```

Comment:
do we really want to hardcode 3rd party URIs into exception messages?

```suggestion
        $"Partial outbox results retrieved with {transportOperationsRetrieved} instead of {expectedNumberOfTransportOperations} transport operation while attempting to load the outbox records for the message id '{messageId}'. This problem can occur due to read-committed isolation levels between transactional write items and query operations. Retrying this message should resolve the issue. For more information consult https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-apis.html#transaction-isolation")
```

Reply:
Yeah there is a risk to that. I'm fine removing it.